homeassistant/components/trest_solar/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for TrestSolarController integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+from .services.trest_identity_service import TrestIdentityService
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate if the user input contains valid credentials."""
+
+    identity = TrestIdentityService(hass, data[CONF_USERNAME], data[CONF_PASSWORD])
+    await identity.authenticate_async()
+
+    if not identity.check_token_is_expired():
```

Comment:
Just to be sure: This returns `False` when the token is expired? (Judging by the name I would assume it returns `True` when the token is expired)